geohisto/models.py

```diff
@@ -1,84 +1,88 @@
         return is_created
 
     def retrieve(self, id):
-        """Get a Town by `id`."""
+        """Get an item by `id`."""
         try:
             return self[id]
         except KeyError:
-            print('Id not found:', id)
+            log.error('Id not found: %s', id)
 
-    def delete(self, town):
-        """Remove a given `town`, do not forget to update references."""
-        del self[town.id]
-
-    def update_successors(self, town, from_town=None, to_town=None):
-        """Update references in case of a Town rename or creation."""
-        try:
-            valid_datetime = town.end_datetime + DELTA
-        except OverflowError:  # Happens on END_DATETIME + DELTA
-            valid_datetime = town.end_datetime
-        if to_town and to_town.valid_at(valid_datetime):
-            self.replace_successor(town, to_town, town.end_datetime)
-        elif from_town and from_town.valid_at(valid_datetime):
-            self.replace_successor(from_town, town)
+    def delete(self, item):
+        """Remove a given `item`, do not forget to update references."""
+        del self[item.id]
 
     def replace_successor(self, old_successor, new_successor,
                           valid_datetime=None):
-        """Run across all Towns and update successors."""
+        """Run across all items and update successors."""
         if valid_datetime is None:
             is_date_valid = True
-        for _town in self.with_successors():
+        for _item in self.with_successors():
             if valid_datetime:
                 try:
-                    is_date_valid = _town.valid_at(valid_datetime)
+                    is_date_valid = _item.valid_at(valid_datetime)
                 except OverflowError:  # Happens on END_DATETIME + DELTA
                     is_date_valid = True
-            if (old_successor.id in _town.successors and
+            if (old_successor.id in _item.successors and
                     is_date_valid and
                     new_successor.start_datetime != START_DATETIME):
-                _town = _town.replace_successor(old_successor.id,
+                _item = _item.replace_successor(old_successor.id,
                                                 new_successor.id)
-                self.upsert(_town)
+                self.upsert(_item)
 
-    def filter(self, **filters):
+    def filter(self, sort=True, **filters):
         """
-        Return a (sorted) list of Town with the given filters applied.
+        Return a (sorted) list of items with the given filters applied.
 
         Useful to look up by `depcom`, `nccenr` and so on.
         """
-        _towns = [town
-                  for town in self.values()
+        _items = [item
+                  for item in self.values()
                   for k, v in filters.items()
-                  if getattr(town, k) == v]
-        _towns.sort()  # Useful for tests.
-        return _towns
+                  if getattr(item, k) == v]
+        if sort:
```

Comment:
Idea: do not sort at all here but sort at the end of the `compute()` method once and for all. It looks cleaner to me and should not affect tests (if that's still the case, let's fix that at the tests level!).